server/src/main/java/org/opensearch/search/SearchPhaseResult.java

```diff
@@ -62,6 +71,8 @@ public abstract class SearchPhaseResult extends TransportResponse {
     protected ShardSearchContextId contextId;
     private ShardSearchRequest shardSearchRequest;
     private RescoreDocIds rescoreDocIds = RescoreDocIds.EMPTY;
+    private TaskResourceInfo taskResourceInfo = new TaskResourceInfo();
```

Comment:
Got some good feedback from .rishabhmaurya , instead of modifying the existing phase level results, we can explore injecting the resource usage into the thread context and piggyback the information with headers.